### PR TITLE
WHO 6 cards

### PR DESCRIPTION
**Pull request by Glorax**
Apologies for accidentally closing the previous one. I am still trying to get the hang of the system. I don't intend to add anything more to this branch.

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Name:Ace, Fearless Rebel
+ManaCost:3 G
+Types:Legendary Creature Human Rebel
+PT:2/2
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigPutCounter | OptionalDecider$ You | TriggerDescription$ Nitro-9 — Whenever CARDNAME attacks, you may sacrifice an artifact. When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:TrigPutCounter:DB$ PutCounter | Cost$ Sac<1/Artifact> | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | AILogic$ Fight | SubAbility$ DBFight
+SVar:DBFight:DB$ Fight | Defined$ ParentTarget | TargetMin$ 0 | TargetMax$ 1 | ValidTgts$ Creature.ControlledBy TriggeredDefendingPlayer | AILogic$ Always | TgtPrompt$ Select target creature defending player controls.
+K:Doctor's companion
+DeckHas:Ability$Counters|Sacrifice & Type$Artifact
+DeckHints:Type$Artifact
```
Comment (Glorax) [inline forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:10]: Would needs be better in this case?

File: forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Name:Ace, Fearless Rebel
+ManaCost:3 G
+Types:Legendary Creature Human Rebel
+PT:2/2
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigPutCounter | OptionalDecider$ You | TriggerDescription$ Nitro-9 — Whenever CARDNAME attacks, you may sacrifice an artifact. When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
```
Comment (Glorax) [inline forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:5]: This ability is currently broken, and I'm not quite sure why.

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:5]: Use AB$ instead of DB$ in line 6. Otherwise the cost will be skipped

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:5]: Also, the oracle says "when you do" which means you will have to use an immediatetrigger

Comment (Glorax) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:5]: Thank you.

File: forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:10]: sure, also the Has Type doesn't fit

File: forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Name:Ace, Fearless Rebel
+ManaCost:3 G
+Types:Legendary Creature Human Rebel
+PT:2/2
+T:Mode$ Attacks | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ TrigPutCounter | OptionalDecider$ You | TriggerDescription$ Nitro-9 — Whenever CARDNAME attacks, you may sacrifice an artifact. When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:TrigPutCounter:AB$ PutCounter | Cost$ Sac<1/Artifact> | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBImmediateTrig | TriggerDescription$ You may sacrifice an artifact. When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:DBImmediateTrig:DB$ ImmediateTrigger | Execute$ DBFight | TriggerDescription$ When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:DBFight:DB$ Fight | Defined$ ParentTarget | TargetMin$ 0 | TargetMax$ 1 | ValidTgts$ Creature.ControlledBy TriggeredDefendingPlayer | TgtPrompt$ Select target creature defending player controls.
```
Comment (Glorax) [inline forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:8]: First half of the ability (sacrificing the artifact and gaining the counter) now works. The fight section of the ability still doesn't work.

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:8]: Try like this
 
```
SVar:DBImmediateTrig:AB$ ImmediateTrigger |  | Cost$ Sac<1/Artifact> | Execute$ TrigPutCounter | TriggerDescription$ When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
SVar:TrigPutCounter:AB$ PutCounter | Cost$ Sac<1/Artifact> | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBFight
SVar:DBFight:DB$ Fight | Defined$ ParentTarget | TargetMin$ 0 | TargetMax$ 1 | ValidTgts$ Creature.ControlledBy TriggeredDefendingPlayer | TgtPrompt$ Select target creature defending player controls.
```

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:8]: This is def. advanced rules territory:
> 603.12. A resolving spell or ability may allow or instruct a player to take an action and create a triggered ability that triggers "when [a player] [does or doesn't]" take that action or "when [something happens] this way." These reflexive triggered abilities follow the rules for delayed triggered abilities (see rule 603.7), except that they're checked immediately after being created and trigger based on whether the trigger event or events occurred earlier during the resolution of the spell or ability that created them. 

1. So the order needs fixing - you pay the cost to get the additional actions.

2. The fight sub has wrong Defined, and the ValidTgts needs `Creature.ControlledBy Spawner>TriggeredDefendingPlayer` which allows you to navigate back to the original trigger and access its triggeringObjects

File: forge-gui/res/cardsfolder/upcoming/graham_o'brien.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Name:Graham O'Brien
+ManaCost:1 G
+Types:Legendary Creature Human Pilot
+PT:2/2
+T:Mode$ SpellCast | ValidCard$ Card.wasNotCastFromYourHand | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigToken | TriggerDescription$ Paradox — Whenever you cast a spell from anywhere other than your hand, create a Food token. (It's an artifact with "{2}, {T}, Sacrifice this artifact: You gain 3 life.")
+SVar:TrigToken:DB$ Token | TokenAmount$ 1 | TokenScript$ c_a_food_sac | TokenOwner$ You
+K:Doctor's companion
+DeckHas:Ability$Token|Sacrifice & Type$Food
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/graham_o'brien.txt:8]: `& Type$Food|Artifact`

File: forge-gui/res/cardsfolder/upcoming/jamie_mccrimmon.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+Name:Jamie McCrimmon
+ManaCost:2 G
+Types:Legendary Creature Human Warrior
+PT:2/2
+K:Trample
+T:Mode$ SpellCast | ValidCard$ Card.Historic | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigPump | TriggerDescription$ Whenever you cast a historic spell, CARDNAME gets +X/+X until end of turn, where X is that spell's mana value. (Artifacts, legendaries, and Sagas are historic.)
+SVar:TrigPump:DB$ Pump | Defined$ Self | NumAtt$ +X | NumDef$ +X
+SVar:X:TriggeredStackInstance$CardManaCostLKI
+K:Doctor's companion
+Oracle:Trample\nWhenever you cast a historic spell, Jamie McCrimmon gets +X/+X until end of turn, where X is that spell's mana value. (Artifacts, legendaries, and Sagas are historic.)\nDoctor's companion (You can have two commanders if the other is the Doctor.)
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/jamie_mccrimmon.txt:10]: add AI line
`DeckHints:Type$Legendary|Saga|Artifact`

**Review by Northmoc — COMMENTED** [review]
Nice PR – hang in there with Git. You'll get the hang of it 😉

File: forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt [hunk 6]
```diff
@@ -0,0 +1,13 @@
+Name:Ace, Fearless Rebel
+ManaCost:3 G
+Types:Legendary Creature Human Rebel
+PT:2/2
+T:Mode$ Attacks | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ DBImmediateTrig | OptionalDecider$ You | TriggerDescription$ Nitro-9 — Whenever CARDNAME attacks, you may sacrifice an artifact. When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:DBImmediateTrig:AB$ ImmediateTrigger | Cost$ Sac<1/Artifact> | Execute$ TrigPutCounter | TriggerDescription$ When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:TrigPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBFight
+SVar:DBFight:DB$ Fight | Defined$ ParentTarget| TargetMin$ 0 | TargetMax$ 1 | ValidTgts$ Creature.ControlledBy Spawner>TriggeredDefendingPlayer | TgtPrompt$ Select target creature defending player controls.
```
Comment (Glorax) [inline forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:8]: The flow of this is infinitely better now, but Ace is still failing to actually fight the targeted creature.

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:8]: Try Defined$ Self

Comment (Glorax) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt]: Works perfectly now.

File: forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+Name:Ace, Fearless Rebel
+ManaCost:3 G
+Types:Legendary Creature Human Rebel
+PT:2/2
+T:Mode$ Attacks | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ DBImmediateTrig | OptionalDecider$ You | TriggerDescription$ Nitro-9 — Whenever CARDNAME attacks, you may sacrifice an artifact. When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:DBImmediateTrig:AB$ ImmediateTrigger | Cost$ Sac<1/Artifact> | Execute$ TrigPutCounter | TriggerDescription$ When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:TrigPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBFight
+SVar:DBFight:DB$ Fight | Defined$ Self| TargetMin$ 0 | TargetMax$ 1 | ValidTgts$ Creature.ControlledBy Spawner>TriggeredDefendingPlayer | TgtPrompt$ Select target creature defending player controls.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt:8]: `Defined$ Self| ` spacing issue
`TgtPrompt$ Select up to one target creature defending player controls` (note no `.`)

Comment (Glorax) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt]: Lothlorien Blade also needs the period removed then.

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt]: Oh yeah, couple things on that script! Good call

**Review by Northmoc — APPROVED** [review]
Nice scripts! 👍

**Pull request merged**